ingesters/customtime/customtime.md

```diff
@@ -0,0 +1,108 @@
+* [File Follower](#!ingesters/ingesters.md#File_Follower)
+* [HTTP Ingester](#!ingesters/ingesters.md#HTTP)
+* [Amazon Kinesis](#!ingesters/ingesters.md#Kinesis_Ingester)
+* [Microsoft Graph API](#!ingesters/ingesters.md#Microsoft_Graph_API_Ingester)
+* [Office 365](#!ingesters/ingesters.md#Office_365_Log_Ingester)
+* [Kafka](#!ingesters/ingesters.md#Kafka)
+
+## Defining a Custom Format
+
+A custom format requires three items to function:
+
+* Name
+* Regular Expression
+* Format
+
+The given name for a custom time format must be unique across other custom time formats and the included timegrinder formats.  For a complete up-to-date listing of included time formats and their names, check out our [timegrinder documentation)[https://pkg.go.dev/github.com/gravwell/gravwell/v3/timegrinder#pkg-constants].
+
+Custom time formats are declared in the configuration files for supported ingesters by specifying a named `TimeFormat` INI block.  Here is an example format which handles timestamps that are delimited using underscores:
+
+```
+[TimeFormat "foo"]
+	Format="2006_01_02_15_04_05"
+	Regex=`\d{4}_\d{1,2}_\d{1,2}_\d{1,2}_\d{1,2}_\d{1,2}`
+```
+
+This format would properly handle the timestamps in the following logs:
+
+```
+2021_02_05_09_00_00 and my id is 1
+2021_02_05_09_00_00 and my id is 2
+2021_02_05_09_00_00 and my id is 3
+2021_02_05_09_00_00 and my id is 4
+2021_02_05_09_00_00 and my id is 5
+2021_02_05_09_00_00 and my id is 6
+```
+
+Here is another format that handles logs with only a timestamp:
+
+```
+[TimeFormat "foo2"]
+	Format="15^04^05"
+	Regex=`\d{1,2}\^\d{1,2}\^\d{1,2}`
+```
+
+This format would handle the following logs, appropriately applying the current date to each extracted timestamp:
+
+```
+09^00^00 and my id is 1
+09^00^00 and my id is 2
+09^00^00 and my id is 3
+09^00^00 and my id is 4
+09^00^00 and my id is 5
+09^00^00 and my id is 6
+```
+
+Note: The custom timestamp format names can be used in [Timestamp-Format-Override](#!ingesters/ingesters.md#Time_Parsing_Overrides) values.
+
+### Time Formats
+
+The `Format` component uses the [golang standard time format specification](https://golang.org/pkg/time/#pkg-constants).  Long story short, you must describe the date `Mon Jan 2 15:04:05 MST 2006` using whatever format you choose.
+
+Time formats can ommit the date component.  When the custom format system identifies that a custom time format does not include a date component it will automatically update the extracted timestamps date to `today`.
+
+Warning: All custom time formats will attempt to operate in UTC unless otherwise indicated using the Format directive.  This means that if you have a time format without a date component you must pay special attention to the timezone.  If an application emits a timestamp of `12:00:00` in MST and there is no timezone component or timezone overrides, timegrinder will interpret the timestamp as UTC and the extracted date will be 7 hours in the past.
```

Comment:
This could use a little clarification. If an application only emits "12:00:00", then you have to set the timezone using the Timezone-Override parameter. If an application emits "12:00:00 EST", you need to make sure you include a timezone in the Format, e.g. "15:04:05 MST".